### DISCUSSION TITLE
property nullable:true missing in OpenAPI description file?

### DISCUSSION THREAD

**Discussion by wi6436** [firefly-iii/firefly-iii#5708] [Firefly III]
I'm using a python API generated by OpenAPI Generator based on the Firefly III OpenAPI description file here: https://github.com/firefly-iii/api-docs/blob/main/firefly-iii-1.5.5.yaml. For details, compare also https://github.com/ms32035/firefly-iii-client.

I had to add the `nullable: true` property to the sections `opening_balance_data` and `UserBlockedCodeProperty` of  the OpenAPI description file at https://github.com/firefly-iii/api-docs/blob/main/firefly-iii-1.5.5.yaml in order to make the OpenAPI generator code work fine.

I'm not sure, if this is a bug in Firefly III.

**Comment by JC5** [comment]
There's no such field as `opening_balance_data`, but all fields called `opening_balance_date` have `nullable: true`.

`UserBlockedCodeProperty` doesn't have `nullable:true` but I always understood that adding `- null` to the list was the same.

**Reply by wi6436** [reply]
Thanks for the reply! You are right, I pasted the wrong field, it's actually `current_debt` and `UserBlockedCodeProperty`, where I added `nullable: true`. `opening_balance_data` doesn't exist and `opening_balance_date` has `nullable: true` already. 

It can very well be that all of this is actually a problem of OpenAPI generator, I'm working with this for the first time and cannot judge it at all; I just wanted to report it.

**Reply by JC5** [reply]
Thanks! I've updated the yaml file! :+1:

**Reply by wi6436** [reply]
Thanks! 👍
